### PR TITLE
Add hinge loss function

### PR DESCRIPTION
**Pull request by shingt**
Added hinge loss function.
Ref: https://github.com/tensorflow/swift-apis/issues/127

### CODE REVIEW

File: Sources/TensorFlow/Loss.swift [hunk 1]
```diff
@@ -36,6 +36,18 @@ public func meanAbsoluteError<Scalar: TensorFlowFloatingPoint>(
     return abs(expected - predicted).mean()
 }
 
+/// Computes the hinge loss between predictions and expectations.
```
Comment (rxwei) [inline Sources/TensorFlow/Loss.swift]:
```suggestion
/// Returns the hinge loss between predictions and expectations.
```

#160 has some more context.

File: Sources/TensorFlow/Loss.swift [hunk 2]
```diff
@@ -36,6 +36,18 @@ public func meanAbsoluteError<Scalar: TensorFlowFloatingPoint>(
...
+///
+/// - Parameters:
+///   - predicted: Predicted outputs from a neural network.
+///   - expected: Expected values, i.e. targets, that correspond to the correct output.
+@differentiable(wrt: predicted)
+public func hinge<Scalar: TensorFlowFloatingPoint>(
```
Comment (rxwei) [inline Sources/TensorFlow/Loss.swift]: I'd suggest renaming this to `hingeLoss(predicted:expected:)`, since "hinge loss" is the canonical name and would be more clear at the call site.

Comment (shingt) [reply Sources/TensorFlow/Loss.swift]: Got it, thanks for your feedback!
